src/transaction/sign.ts

```diff
@@ -55,12 +49,61 @@ function signWithKeypair(
   }
 
   const serialized = binary.encode(tx)
+
+  // Decode the serialized transaction:
+  const decoded = binary.decode(serialized)
+
+  // ...And ensure it is equal to the original tx, except:
+  // - It must have a TxnSignature or txSigners (multisign).
+  if (!decoded.TxnSignature && !tx.Signers) {
+    throw new utils.common.errors.ValidationError(
+      'Serialized signed transaction is missing "TxnSignature" property'
```

Comment:
Consider rephrasing to line up with the check you're actually performing:
"Serialized transaction must have a TxnSignature or Signers property."

Reply:
Not done